class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+            $this->get_plugin_setting(self::GF_RAZORPAY_SECRET));
+
+        $attributes = array (
+            'razorpay_order_id'   => $razorpay_order_id,
+            'razorpay_payment_id' => rgpost('razorpay_payment_id'),
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if ($entry and !empty(rgpost('razorpay_payment_id')) and !empty(rgpost('razorpay_signature')))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (Errors\SignatureVerificationError $e)
+            {
+                $error = $e->getMessage();
+            }
+        }
+        else
+        {
+            $error = 'Payment Failed';
+        }
+
+        if ($success === true)
+        {
+            $action['id']             = $attributes['razorpay_payment_id'];
+            $action['type']           = 'complete_payment';
+            $action['transaction_id'] = $attributes['razorpay_payment_id'];
+            $action['amount']         = $entry['payment_amount'];
+            $action['entry_id']       = $entry['id'];
+        }
+        else
+        {
+            $action['id']             = $attributes['razorpay_payment_id'];
+            $action['type']           = 'fail_payment';
+            $action['transaction_id'] = $attributes['razorpay_payment_id'];
+            $action['entry_id']       = $entry['id'];
+            $action['amount']         = $entry['payment_amount'];
+        }
+
+        return $action;
+    }
+
+    public function post_callback($callback_action, $callback_result) {
+        $entry               = GFAPI::get_entry($callback_action['entry_id']);
+        $feed                = $this->get_payment_feed( $entry );
+        $transaction_id      = rgar($callback_action, 'transaction_id');
+        $amount              = rgar($callback_action, 'amount');
+
+        do_action('gform_razorpay_post_payment_' . $callback_action['type'],  $_POST, $entry, $feed);
+    }
+
+    public function generate_razorpay_form($entry, $customer_fields, $form)
+    {
+        $feed = $this->get_payment_feed($entry, $form);
+
+        $razorpay_args = array(
+            'key'         => $this->get_plugin_setting(self::GF_RAZORPAY_KEY),
+            'name'        => $form['name'],
+            'amount'      => (int) round($entry['payment_amount']*100),
+            'currency'    => $entry['currency'],
+            'description' => $form['description'],
+            'prefill'     => array(
+                'name'    => $customer_fields['billingInformation_firstName'],
+                'email'   => $customer_fields['billingInformation_email'],
+            ),
+            'notes'       => array(
+                'gravity_forms_order_id' => $entry['id']
+            ),
+            'order_id'    => $entry['razorpay_order_id'],
+        );
+
+        $json = $razorpay_args;
```

Comment:
drop the assignment and use `$razorpay_args` (should be `$razorpayArgs`?) directly instead below?